java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -1136,38 +930,21 @@ public CatchTreeImpl newCatchClause(JavaTree catchToken, InternalSyntaxToken ope
   }
 
   public VariableTreeImpl newCatchFormalParameter(Optional<ModifiersTreeImpl> modifiers, TypeTree type, VariableTreeImpl parameter) {
-    // TODO modifiers
-
-    if (modifiers.isPresent()) {
-      parameter.prependChildren(modifiers.get(), (AstNode) type);
-    } else {
-      parameter.prependChildren((AstNode) type);
-    }
-
+    // FIXME SONARJAVA-547 keep annotations from modifiers
     return parameter.completeType(type);
   }
 
-  public TypeTree newCatchType(TypeTree qualifiedIdentifier, Optional<List<AstNode>> rests) {
+  public TypeTree newCatchType(TypeTree qualifiedIdentifier, Optional<List<Tuple<InternalSyntaxToken, TypeTree>>> rests) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)